cmd/krel/cmd/sign_blobs.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/nozzle/throttler"
+	"github.com/spf13/cobra"
+
+	sign "sigs.k8s.io/release-sdk/sign"
+)
+
+// signBlobCmd represents the subcommand for `krel sign blobs`
+var signBlobCmd = &cobra.Command{
+	Use:           "blobs",
+	Short:         "Sign blobs",
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runSignBlobs(singOpts, args)
+	},
+}
+
+func init() {
+	signCmd.AddCommand(signBlobCmd)
+}
+
+func runSignBlobs(signOpts *signOptions, args []string) error {
+	if err := signOpts.Validate(); err != nil {
+		return fmt.Errorf("validating sign blobs options: %w", err)
+	}
+
+	err := validateSignBlobsArgs(args)
+	if err != nil {
+		return fmt.Errorf("blobs to be signed does not exist: %w", err)
+	}
+
+	t := throttler.New(int(signOpts.maxWorkers), len(args))
+	for _, file := range args {
+		go func(file string) {
+			fileName := strings.TrimSuffix(file, filepath.Ext(file))
+
+			signerOpts := sign.Default()
+			signerOpts.Verbose = signOpts.verbose
+			signerOpts.Timeout = signOpts.timeout
+			signerOpts.PrivateKeyPath = signOpts.privateKeyPath
+			signerOpts.PublicKeyPath = signOpts.publicKeyPath
+			signerOpts.OutputCertificatePath = fmt.Sprintf("%s/%s.pem", signOpts.outputCertificatePath, fileName)
+			signerOpts.OutputSignaturePath = fmt.Sprintf("%s/%s.sig", signOpts.outputCertificatePath, fileName)
+
+			signer := sign.New(signerOpts)
+			_, err := signer.SignFile(file)
+			if err != nil {
+				t.Done(fmt.Errorf("signing the file %s: %w", file, err))
+				return
+			}
+
+			t.Done(nil)
+		}(file)
+
+		if t.Throttle() > 0 {
+			break
+		}
+	}
+
+	if err := t.Err(); err != nil {
+		return fmt.Errorf("signing the blobs: %w", err)
+	}
+
+	return nil
+}
+
+func validateSignBlobsArgs(args []string) error {
+	for _, file := range args {
+		if _, err := os.Stat(file); os.IsNotExist(err) {
+			return fmt.Errorf("blob %s does not exist", file)
+		}
+	}
+
+	return nil
+}
```

Comment:
Do we need to check that? I assume the sign API will error if the file does not exist.

Reply:
maybe not, just to fail fast in this case